cfme/infrastructure/virtual_machines.py

```diff
@@ -1666,17 +1688,34 @@ def step(self, *args, **kwargs):
 
 @navigator.register(InfraTemplateCollection, 'TemplatesOnly')
 class TemplatesAll(CFMENavigateStep):
-    VIEW = TemplatesOnlyAllView
-    prerequisite = NavigateToSibling('All')
+
```

Comment:
Check out how `InfraTemplateCollection.all` method is changing its navigation destination based on the filter. The changes you're making to this navigation class are actually crossing over with the existing `ProviderTemplates` destination that's registered against the provider instance.

ComparableMixin could pick its `navigate_to` call based on the parent/filter, instead of trying to push that logic down here into the navigation destination.

Reply:
Just to be clear, this is to be used instead of 
```
if self.obj.parent is self.obj.appliance:
            nav_inst = self.obj
            nav_dest = 'All'
        else:
            nav_inst = self.obj.parent
            nav_dest = 'Details'
        return navigate_to(nav_inst, nav_dest)``` 
Makes sense, and we can leave the nav destination as is was.

Reply:
I implemented NAV_STRING as a property in the Collection classes and had it pass the appropriate string based upon the existence of a provider filter.